### PR TITLE
Selenium: cover Golang language server functionality by selenium test 

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
This PR creates **GolangFileEditingTest** selenium test that cover the **Golang** language server functionality by use cases from **Language servers** test plan - https://github.com/eclipse/che/wiki/Test-plan-for-checking-Language-servers.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10084

### CODE REVIEW

**Comment by SkorikSergey** [comment]
ci-build

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    editor.waitTextIntoAutocompleteContainer("Println");
+    editor.waitTextIntoAutocompleteContainer("Printf");
+
+    editor.deleteCurrentLine();
+    editor.waitAllMarkersInvisibility(ERROR);
+  }
+
+  @Test(priority = 1)
+  public void checkCodeValidation() {
+    editor.selectTabByName(GO_FILE_NAME);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java]: From the readability view it's better to divide separate steps by empty rows.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    editor.waitAllMarkersInvisibility(ERROR);
+    editor.goToCursorPositionVisible(13, 1);
+    editor.typeTextIntoEditor("p");
+    editor.waitMarkerInPosition(ERROR, 13);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java]: IMHO it's important to check error message as well.

**Review by dmytro-ndp — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    URL resource = ApacheCamelFileEditingTest.class.getResource("/projects/go-simple");
+    testProjectServiceClient.importProject(
+        workspace.getId(), Paths.get(resource.toURI()), PROJECT_NAME, ProjectTemplates.GO);
+
+    ide.open(workspace);
+  }
+
+  @Test
+  public void checkLanguageServerInitialized() {
+    projectExplorer.waitAndSelectItem(PROJECT_NAME);
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java]:
```
    projectExplorer.waitAndSelectItem(PROJECT_NAME);
    projectExplorer.openItemByPath(PROJECT_NAME);
    projectExplorer.openItemByPath(PATH_TO_GO_FILE);
```
Can be changed to "ProjectExplorer#expandPathInProjectExplorerAndOpenFile(String path, String fileName)"

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    editor.selectTabByName(GO_FILE_NAME);
+
+    // launch autocomplete feature and check proposals list
+    editor.goToPosition(21, 58);
+    editor.typeTextIntoEditor(Keys.ENTER.toString());
+    editor.typeTextIntoEditor("fmt.");
+    editor.launchAutocompleteAndWaitContainer();
+
+    editor.checkProposalDocumentation("No documentation found.");
+    editor.waitTextIntoAutocompleteContainer("Print");
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java]: maybe we should works this method for using of multiple arguments:
```
CodenvyEditor#waitTextIntoAutocompleteContainer(String ... expectedText);
```
It helps to reach more compact syntax:
```
editor.waitTextIntoAutocompleteContainer("Print", "Println",  "Printf");
```
what do you think about this?

**Review by Ohrimenko1988 — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java (hunk 3 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/GolangFileEditingTest.java]: Can we apply quickExpandWithJavaScript()? For reducing of opening time.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**